packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -126,14 +132,27 @@ export default class WorkspacesForeachCommand extends BaseCommand {
       ? project.topLevelWorkspace
       : cwdWorkspace!;
 
+    if (configuration.projectCwd === null)
+      throw new UsageError(`This command can only be run from within a Yarn project`);
+
+    const root = await gitUtils.fetchRoot(configuration.projectCwd);
+    const base = root !== null
+      ? await gitUtils.fetchBase(root, {baseRefs: [typeof this.since === `string` ? this.since : await gitUtils.fetchDefaultBranch(root)]})
+      : null;
```

Comment:
Same thing: better not call git unless we need to.

Reply:
Fixed in 0fcfcbea37bbd13b1d08ffd1359ce8fc12d96692.